flink-cdc-connect/flink-cdc-pipeline-connectors/flink-connector-test-cdc/src/main/java/com/ververica/cdc/connectors/test/sink/ValuesDataSink.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2023 Ververica Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ververica.cdc.connectors.test.sink;
+
+import org.apache.flink.api.connector.sink2.SinkWriter;
+
+import com.ververica.cdc.common.event.DataChangeEvent;
+import com.ververica.cdc.common.event.Event;
+import com.ververica.cdc.common.sink.DataSink;
+import com.ververica.cdc.common.sink.EventSinkProvider;
+import com.ververica.cdc.common.sink.FlinkSinkProvider;
+import com.ververica.cdc.common.sink.MetadataApplier;
+import com.ververica.cdc.connectors.test.ValuesDatabase;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+
+/** A {@link DataSink} for "values" connector that supports schema evolution. */
+public class ValuesDataSink implements DataSink, Serializable {
+
+    /** Accumulate a certain number of {@link Event} to trigger flush. */
+    private final int flushSize;
+
+    public ValuesDataSink(int flushSize) {
+        this.flushSize = flushSize;
+    }
+
+    @Override
+    public EventSinkProvider getEventSinkProvider() {
+        return FlinkSinkProvider.of((initContext) -> new TestValuesSinkWriter(flushSize));
+    }
+
+    @Override
+    public MetadataApplier getMetadataApplier() {
+        return new ValuesDatabase.ValuesMetadataApplier();
+    }
+
+    private static class TestValuesSinkWriter implements SinkWriter<Event> {
+
+        private final List<DataChangeEvent> cache = new ArrayList<>();
+
+        /** Accumulate a certain number of {@link Event} to trigger flush. */
+        private final int flushSize;
```

Comment:
The parameter looks like used to do batch or buffer work, should we rename it to batchSize or bufferSize to avoid conflict with `FlushEvent`?